biz.aQute.bndlib/src/aQute/bnd/differ/Baseline.java

```diff
@@ -136,6 +134,8 @@ public Set<Info> baseline(Tree n, Parameters nExports, Tree o, Parameters oExpor
 			if (!packageFilters.matches(pdiff.getName()))
```

Comment:
Not really happy how you change the code I gave you. This is less efficient (you search the Instructions twice now) and though Optional can be nice in APIs, this is not a proper use case imho. You get multiple boxing/unboxing and a simple comparison becomes numerous calls with intermediate objects.
It is not a major deal here but in a tool like bnd you'd like to keep operations that are called often as efficient as possible. And this does not even increase readability by adding more code.

Reply:
I find your approach less intelligible because `getThreshold()` suddenly has an implicit side-effect, namely that if there is no threshold, there is no rule. And for some reason I find difficult to trace down, that logic is broken as evidenced by the fact that when I use your code as-is, it breaks a bunch of test cases in `BaselineTest`.

I'll remove the optional and arrange the code slightly differently.